legend-engine-pure-code-compiled-core/src/main/resources/core/pure/router/preeval/tests.pure

```diff
@@ -1404,6 +1404,61 @@ function <<test.Test>> meta::pure::router::preeval::tests::testFilterNoSimplific
   assertRoundTrip($input, $expected);
 }
 
+
+function <<test.Test>> meta::pure::router::preeval::tests::testToOneManyElimination1():Boolean[1]
+{
+  let input = {| 'hello'->toOneMany()};
+
+  let expected = {| 'hello'};
+
+  assertRoundTrip($input, $expected);
+}
+
+function <<test.Test>> meta::pure::router::preeval::tests::testToOneManyElimination2():Boolean[1]
+{
+  let input = {x:String[*]| $x->toOneMany()};
+
+  let expected = $input;
+
+  assertRoundTrip($input, $expected);
+}
+
+function <<test.Test>> meta::pure::router::preeval::tests::testToOneManyElimination3():Boolean[1]
+{
+  let input = {x:String[1..*]| $x->toOneMany()};
+
+  let expected = {x:String[1..*]| $x};
+
+  assertRoundTrip($input, $expected);
+}
+
+function <<test.Test>> meta::pure::router::preeval::tests::testToOneElimination1():Boolean[1]
+{
+  let input = {| 'hello'->toOne()};
+
+  let expected = {| 'hello'};
+
+  assertRoundTrip($input, $expected);
+}
+
+function <<test.Test>> meta::pure::router::preeval::tests::testToOneElimination2():Boolean[1]
+{
+  let input = {x:String[*]| $x->toOne()};
+
+  let expected = $input;
+
+  assertRoundTrip($input, $expected);
+}
+
+function <<test.Test>> meta::pure::router::preeval::tests::testToOneElimination3():Boolean[1]
+{
+  let input = {x:String[1]| $x->toOne()};
+
+  let expected = {x:String[1..*]| $x};

```

Comment:
How is this correct expected?  We went from [1], to [1..*].  That seems to change the semantics of the function completely.  Like, on relational, can this lead to an in clauses, for example?

Reply:
The expected function signature is wrong, it should be [1] - then the toOne is not needed.  (Fixing in another MR)